File: exercises/domain-5/block-5.4/starter.py
import anthropic
import json

client = anthropic.Anthropic()


def search_web(query: str) -> list[dict]:
    """Simulated search results"""
    results = {
        "claude context window": [
            {"url": "https://docs.anthropic.com/claude/context-window", "snippet": "Claude 3.5 Sonnet supports up to 200,000 tokens."},
        ],
        "anthropic pricing": [
            {"url": "https://anthropic.com/pricing", "snippet": "Claude API pricing varies by model."},
        ],
    }
    return results.get(query.lower(), [{"url": "https://anthropic.com", "snippet": "Anthropic is an AI safety company."}])


def research_with_provenance(question: str) -> str:
    """TODO: Research with provenance tracking and source citations"""
    return f'TODO: Research "{question}" with provenance tracking'


print(research_with_provenance("What is Claude's context window size?"))

File: exercises/domain-5/block-5.4/starter.ts
import Anthropic from "@anthropic-ai/sdk"

const client = new Anthropic()

// Simulated search results with sources
function searchWeb(query: string): { url: string; snippet: string }[] {
  const results: Record<string, { url: string; snippet: string }[]> = {
    "claude context window": [
      { url: "https://docs.anthropic.com/claude/context-window", snippet: "Claude 3.5 Sonnet supports up to 200,000 tokens context window." },
      { url: "https://anthropic.com/news/claude-3", snippet: "Anthropic's Claude 3 models feature extended context windows." },
    ],
    "anthropic pricing": [
      { url: "https://anthropic.com/pricing", snippet: "Claude API pricing varies by model. Haiku is the most cost-effective option." },
    ],
  }
  return results[query.toLowerCase()] ?? [{ url: "https://anthropic.com", snippet: "Anthropic is an AI safety company." }]
}

// TODO: Implement provenance-tracking research
// The output should have inline citations and a sources list
// Training knowledge claims should be explicitly flagged
async function researchWithProvenance(question: string): Promise<string> {
  return `TODO: Research "${question}" with provenance tracking`
}

researchWithProvenance("What is Claude's context window size?")
  .then(console.log)
  .catch(console.error)
